File: src/main/java/com/euc/grantfitassessment/eval/FitReasonerVariant.java
package com.euc.grantfitassessment.eval;

import com.euc.grantfitassessment.FitReasoner;

/**
 * One configuration under test in the drift experiment (docs/proposal.md
 * Section 7): a FitReasoner implementation (a specific model + prompt)
 * plus whether this change is expected — independently of the experiment's
 * outcome — to alter reasoning behavior. That expectation is the "known to
 * alter behavior" comparison point Section 7 step 5 measures detection
 * against, so it must be set deliberately by whoever designs the variant
 * (e.g. a genuinely different model = true, a harmless refactor of prompt
 * wording that shouldn't change meaning = false), not inferred from
 * DriftExperimentRunner's results.
 */
public record FitReasonerVariant(String label, FitReasoner reasoner, boolean expectedToAlterBehavior) {
}

File: src/euc/grantfitassessment/eval/fit_reasoner_variant.py
from __future__ import annotations

from dataclasses import dataclass

from euc.grantfitassessment.reasoner import FitReasoner


@dataclass(frozen=True)
class FitReasonerVariant:
    """A labeled configuration triple: display label, the FitReasoner
    implementation, and whether this variant is *expected* to change
    behavior — the ground truth the drift metrics are computed against."""

    label: str
    reasoner: FitReasoner
    expected_to_alter_behavior: bool
